### PR TITLE
Feature/upload api from filesystem

### PR DESCRIPTION
**Pull request by sudhirtumati**

### CODE REVIEW

File: src/main/java/io/apicollab/uploader/fs/FileWatcher.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        WatchKey watchKey;
+        log.info("Scanning filesystem for changes");
+        log.info("Place api specs in '{}' folder to upload them to portal", path.toAbsolutePath().toString());
+        while ((watchKey = watchService.take()) != null) {
+            for (WatchEvent<?> event : watchKey.pollEvents()) {
+                log.info("File: {} - {}", event.context(), event.kind());
+                String filename = event.context().toString();
+                log.debug("Processing file: {}", filename);
+                try {
+                    String spec = loadSpec(filename);
```
Comment (kand617) [inline src/main/java/io/apicollab/uploader/fs/FileWatcher.java:49]: Does loading the spec automatically validate the spec too?

File: README.md [hunk 2]
```diff
@@ -1,3 +1,32 @@
...
+mvn clean package
+` 
+# Deploy
+`
+java -jar target\fs-uploader.jar
+`
+# Configuration
+## File upload directory
+`${java.io.tmpdir}/api_input` folder is scanned for changes by default. This can be changed with `upload.baseDir` command line argument.
+## Api portal base url
```
Comment (kand617) [inline README.md:35]: Is it worth mentioning the required vendor extensions in the swagger file?
x-application-name and any other ones

**Review by kand617 — CHANGES_REQUESTED** [review]

File: src/main/java/io/apicollab/uploader/fs/FileWatcher.java (hunk 1 above)
Comment (sudhirtumati) [reply src/main/java/io/apicollab/uploader/fs/FileWatcher.java:49]: At the moment, uploading a spec from portal and from fs-uploader results in same behavior. In my view, uploading a spec doesn't mean the spec is validated. Validation and publishing must be an extra step to be performed from within portal .

@kand617 What is your opinion?

File: README.md (hunk 2 above)
Comment (sudhirtumati) [reply README.md:35]: Done

File: src/main/java/io/apicollab/uploader/fs/FileWatcher.java (hunk 1 above)
Comment (kand617) [reply src/main/java/io/apicollab/uploader/fs/FileWatcher.java:49]: OK agreed.

File: README.md [hunk 3]
```diff
@@ -3,12 +3,24 @@
...
 - Git
-
+# Mandatory custom property in Swagger/OpenAPI spec
+Every API spec must be linked to an existing application. For the uploader module to upload API spec under specific application, a custom property `x-app-id` must be made available under `info` section of spec to be uploaded. Refer to sample snippet below:
+```
+openapi: "3.0.0"
+info:
+  version: 1.0.0
+  title: API Collab - Server
+  x-app-id: 1 # Custom property for application id
```
Comment (kand617) [inline README.md:17]: Its more user friendly to call it x-app-name and have the name of the application?

**Review by kand617 — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (sudhirtumati) [reply README.md:17]: Agree but this might need few changes on portal's API.

Created #2 for this change.

**Pull request merged**